libs/alerts/src/main/java/org/codice/ddf/system/alerts/SystemNotice.java

```diff
@@ -0,0 +1,175 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.system.alerts;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+
+/**
+ * SystemNotice is the property map class used by DDF collectors when sending system notifications to
+ * the Decanter topic 'decanter/collect/*'. This class defines all the common keys used in system
+ * notification events and provides convenient helpers for accessing them.
+ */
+public class SystemNotice extends HashMap<String, Object> {
+    //topics
+    public static final String SYSTEM_NOTICE_BASE_TOPIC = "decanter/collect/";
+
+    public static final String SYSTEM_NOTICE_EVENT_TYPE = "system-notice";
+
+    //common decanter fields
+    //Key for a type identifier. Included to be consistent with other decanter events. Will always be system-notice
+    public static final String EVENT_TYPE_KEY = "type";
+
+    //Key for the host name on which this notice originated
+    public static final String SYSTEM_NOTICE_HOST_NAME_KEY = "hostName";
+
+    //Key for the host address on which this notice originated
+    public static final String SYSTEM_NOTICE_HOST_ADDRESS_KEY = "hostAddress";
+
+    //custom system notice fields
+    //Key for the notice id. This is an auto generated unique id
+    public static final String SYSTEM_NOTICE_ID_KEY = "id";
+
+    //Key for the date/time of this notice. This is auto populated on creation
+    public static final String SYSTEM_NOTICE_TIME_KEY = "noticeTime";
+
+    //Key for the source of this notice. Usually this is a class name and possibly followed by an additional discriminator (ie method name)
+    public static final String SYSTEM_NOTICE_SOURCE_KEY = "source";
+
+    //Key for the priority of the notice. Acceptable values can be found in NoticePriority
+    public static final String SYSTEM_NOTICE_PRIORITY_KEY = "priority";
+
+    //Key for the title of the notice. Defaults to an empty string
+    public static final String SYSTEM_NOTICE_TITLE_KEY = "title";
+
+    //Key for the Set<String> of details of this notice
```

Comment:
❓ Any special expectations for details? Will they be used for display purposes? Why a collection instead of a simple `String`?